### PR TITLE
replace assert_se(streq/streq_ptr) with ASSERT_STREQ for simple cases

### PR DESCRIPTION
**Pull request by ikruglov**
As subject describes

### CODE REVIEW

File: src/test/test-open-file.c [hunk 1]
```diff
@@ -179,7 +179,7 @@ TEST(open_file_to_string) {
         r = open_file_to_string(of, &s);
 
         assert_se(r >= 0);
-        assert_se(streq(s, "/path\\:with\\:colon"));
+        ASSERT_STREQ(s, "/path\\:with\\:colon");
```
Comment (YHNdnzj) [inline src/test/test-open-file.c]: Please drop this. Will conflict with #32181.

Comment (ikruglov) [reply src/test/test-open-file.c]: dropped

**Review by DaanDeMeyer — APPROVED** [review]

File: src/test/test-string-util.c [hunk 2]
```diff
@@ -163,128 +163,128 @@ TEST(ascii_strcasecmp_nn) {
...
+        ASSERT_STREQ(cellescape(buf, 6, "\a\b\f\n\r\t\v\\\"'"), is_locale_utf8() ? "\\a…" : "\\a...");
+        ASSERT_STREQ(cellescape(buf, 7, "\a\b\f\n\r\t\v\\\"'"), is_locale_utf8() ? "\\a…" : "\\a...");
+        ASSERT_STREQ(cellescape(buf, 8, "\a\b\f\n\r\t\v\\\"'"), is_locale_utf8() ? "\\a\\b…" : "\\a\\b...");
+
+        ASSERT_STREQ(cellescape(buf, sizeof buf, "1\020"), "1\\020");
+        ASSERT_STREQ(cellescape(buf, sizeof buf, "1\020x"), "1\\020x");
 }
 
 TEST(streq_ptr) {
-        assert_se(streq_ptr(NULL, NULL));
```
Comment (mrc0mmand) [inline src/test/test-string-util.c]: IMO this shouldn't be changed, as it's supposed to explicitly check the `streq_ptr()` function.

**Comment by mrc0mmand** [comment]
Also, there's maybe one potential issue(?)/implementation detail - this replaces calls to both `streq()` _and_ `streq_ptr()` with `streq_ptr()` (hidden behind `ASSERT_STREQ()`). I'm not completely sure why we still differentiate between the two, but I suspect that's something to do with the former not handling `NULL` properly and crashing, so it serves as another check for programming errors(?). With this patch we'd lose this part. But I'm not even sure if that's a real concern, just something that came to my mind while reviewing the patch.

**Comment by DaanDeMeyer** [comment]
I think for the assertion macro it's fine to always use streq_ptr(), it'll fail with a more descriptive error if one of the arguments is `NULL` compared to crashing in `streq`.

**Pull request merged**